substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/NativeImageInlineDuringParsingPlugin.java

```diff
@@ -190,10 +193,22 @@ public NativeImageInlineDuringParsingPlugin(boolean analysis, HostedProviders pr
 
     @Override
     public InlineInfo shouldInlineInvoke(GraphBuilderContext b, ResolvedJavaMethod method, ValueNode[] args) {
+
+        if (Snippets.class.isAssignableFrom(((AnalysisMethod) (b.getMethod())).getDeclaringClass().getJavaClass())) {
+            /* We are not interfering with any snippet */
+            // System.out.println(((AnalysisMethod) (b.getMethod())).getDeclaringClass().getJavaClass());
+            return null;
+        }
+
+        if (Snippets.class.isAssignableFrom(((AnalysisMethod)method).getDeclaringClass().getJavaClass())) {
+            /* We are not interfering with any snippet */
+            // System.out.println(((AnalysisMethod)method).getDeclaringClass().getJavaClass());
+            return null;
+        }
+
```

Comment:
> `doArraycopy` is a snippet and it should never end up being inlined in the first place. We have to filter it earlier.
@vjovanov is it ok to filter snippets this way?

Reply:
after adding this filter, null InvocationData continues to be a problem for some methods from packages com.oracle.svm.core.snippets and com.oracle.svm.core.genscavenge  - can I filter them the same way?